cumulus/parachains/runtimes/coretime/coretime-rococo/src/coretime.rs

```diff
@@ -0,0 +1,298 @@
+				e
+			),
+		}
+	}
+
+	fn credit_account(who: Self::AccountId, amount: Self::Balance) {
+		use crate::coretime::CoretimeProviderCalls::CreditAccount;
+		let credit_account_call = BrokerRuntimePallets::<
+			CoreIndex,
+			BlockNumber,
+			AccountId,
+			Balance,
+			CoreAssignment,
+			PartsOf57600,
+		>::Broker(CreditAccount(who, amount));
+
+		let message = Xcm(vec![
+			Instruction::UnpaidExecution {
+				weight_limit: WeightLimit::Unlimited,
+				check_origin: None,
+			},
+			Instruction::Transact {
+				origin_kind: OriginKind::Native,
+				require_weight_at_most: Weight::from_parts(1, 1),
+				call: credit_account_call.encode().into(),
+			},
+		]);
+
+		match PolkadotXcm::send_xcm(Here, MultiLocation::parent(), message.clone()) {
+			Ok(_) => log::info!(
+				target: "runtime::coretime",
+				"Instruction to credit account sent successfully."
+			),
+			Err(e) => log::error!(
+				target: "runtime::coretime",
+				"Instruction to credit account failed to send: {:?}",
+				e
+			),
+		}
+	}
+
+	fn assign_core(
+		core: CoreIndex,
+		begin: Self::BlockNumber,
+		assignment: Vec<(CoreAssignment, PartsOf57600)>,
+		end_hint: Option<Self::BlockNumber>,
+	) {
+		use crate::coretime::CoretimeProviderCalls::AssignCore;
+		let assign_core_call = BrokerRuntimePallets::<
+			CoreIndex,
+			BlockNumber,
+			AccountId,
+			Balance,
+			CoreAssignment,
+			PartsOf57600,
+		>::Broker(AssignCore(core, begin, assignment, end_hint));
+
+		let message = Xcm(vec![
+			Instruction::UnpaidExecution {
+				weight_limit: WeightLimit::Unlimited,
+				check_origin: None,
+			},
+			Instruction::Transact {
+				origin_kind: OriginKind::Native,
+				require_weight_at_most: Weight::from_parts(1, 1),
+				call: assign_core_call.encode().into(),
+			},
+		]);
+
+		match PolkadotXcm::send_xcm(Here, MultiLocation::parent(), message.clone()) {
+			Ok(_) => log::info!(
+				target: "runtime::coretime",
+				"Core assignment sent successfully."
+			),
+			Err(e) => log::error!(
+				target: "runtime::coretime",
+				"Core assignment failed to send: {:?}",
+				e
+			),
+		}
+	}
+
+	fn check_notify_core_count() -> Option<u16> {
+		let count = CoreCount::get();
+		CoreCount::set(&None);
+		count
+	}
+
+	fn check_notify_revenue_info() -> Option<(Self::BlockNumber, Self::Balance)> {
+		let revenue = CoretimeRevenue::get();
+		CoretimeRevenue::set(&None);
+		revenue
+	}
+
+	#[cfg(feature = "runtime-benchmarks")]
+	fn ensure_notify_core_count(count: u16) {
+		CoreCount::set(&Some(count));
+	}
+
+	#[cfg(feature = "runtime-benchmarks")]
+	fn ensure_notify_revenue_info(when: Self::BlockNumber, revenue: Self::Balance) {
+		CoretimeRevenue::set(&Some((when, revenue)));
+	}
+}
+
+impl pallet_broker::Config for Runtime {
+	type RuntimeEvent = RuntimeEvent;
+	type Currency = Balances;
+	type OnRevenue = CreditToStakingPot<Runtime>;
+	type TimeslicePeriod = ConstU32<2>;
+	type MaxLeasedCores = ConstU32<5>;
+	type MaxReservedCores = ConstU32<5>;
```

Comment:
Asset Hub, Bridge Hub, Collectives, Coretime, People. So it's just enough. But this is a maximum, is it possible (or desirable) to have it be more so that we don't need an upgrade of this chain to add another system chain?

Of course if it actually reserves a core, that's not very efficient, so we should set it to the exact number needed. But if it's just a ceiling, then perhaps 10?

Reply:
I just checked and it is a ceiling, cores are then reserved by an admin origin or root. 10 sounds reasonable.